--- dropbear-2025.89/src/svr-auth.c
+++ dropbear-2025.89/src/svr-auth.c
@@ -427,7 +427,7 @@ void send_msg_userauth_failure(int partial, int incrfail) {
 		ses.authstate.failcount++;
 	}
 
-	if (ses.authstate.failcount >= svr_opts.maxauthtries) {
+	if (ses.authstate.failcount > svr_opts.maxauthtries) {
 		char * userstr;
 		/* XXX - send disconnect ? */
 		TRACE(("Max auth tries reached, exiting"))
--- dropbear-2025.89/src/svr-runopts.c
+++ dropbear-2025.89/src/svr-runopts.c
@@ -426,8 +426,7 @@ void svr_getopts(int argc, char ** argv) {
 
 	if (maxauthtries_arg) {
 		unsigned int val = 0;
-		if (m_str_to_uint(maxauthtries_arg, &val) == DROPBEAR_FAILURE 
-			|| val == 0) {
+		if (m_str_to_uint(maxauthtries_arg, &val) == DROPBEAR_FAILURE) {
 			dropbear_exit("Bad maxauthtries '%s'", maxauthtries_arg);
 		}
 		svr_opts.maxauthtries = val;
