--- dropbear-2025.89/src/svr-chansession.c
+++ dropbear-2025.89/src/svr-chansession.c
@@ -438,6 +438,11 @@ static int sessionsignal(const struct ChanSess *chansess) {
 		return DROPBEAR_FAILURE;
 	}
 
+	if (svr_opts.forced_command || svr_pubkey_has_forced_command()) {
+		TRACE(("disallowed signal for forced_command"));
+		return DROPBEAR_FAILURE;
+	}
+
 	signame = buf_getstring(ses.payload, NULL);
 
 	for (i = 0; signames[i].name != NULL; i++) {
