diff -uNr chrony-4.5/sys_linux.c chrony-4.5.new/sys_linux.c
--- chrony-4.5/sys_linux.c	2023-12-05 08:22:10.000000000 -0500
+++ chrony-4.5.new/sys_linux.c	2024-04-24 00:01:15.579026838 -0400
@@ -485,7 +485,6 @@
 #endif
     SCMP_SYS(gettimeofday),
     SCMP_SYS(settimeofday),
-    SCMP_SYS(time),
 
     /* Process */
     SCMP_SYS(clone),
@@ -495,7 +494,6 @@
     SCMP_SYS(exit),
     SCMP_SYS(exit_group),
     SCMP_SYS(getpid),
-    SCMP_SYS(getrlimit),
     SCMP_SYS(getuid),
     SCMP_SYS(getuid32),
 #ifdef __NR_membarrier
@@ -515,7 +513,6 @@
     /* Memory */
     SCMP_SYS(brk),
     SCMP_SYS(madvise),
-    SCMP_SYS(mmap),
     SCMP_SYS(mmap2),
     SCMP_SYS(mprotect),
     SCMP_SYS(mremap),
@@ -575,8 +572,6 @@
     SCMP_SYS(sendmsg),
     SCMP_SYS(sendto),
     SCMP_SYS(shutdown),
-    /* TODO: check socketcall arguments */
-    SCMP_SYS(socketcall),
 
     /* General I/O */
     SCMP_SYS(_newselect),
@@ -600,7 +595,6 @@
 #ifdef __NR_futex_time64
     SCMP_SYS(futex_time64),
 #endif
-    SCMP_SYS(select),
     SCMP_SYS(set_robust_list),
     SCMP_SYS(write),
     SCMP_SYS(writev),
@@ -609,6 +603,15 @@
     SCMP_SYS(getrandom),
     SCMP_SYS(sysinfo),
     SCMP_SYS(uname),
+    /* not always available */
+#if ! defined(__ARM_EABI__)
+    SCMP_SYS(time),
+    SCMP_SYS(getrlimit),
+    SCMP_SYS(select),
+    SCMP_SYS(mmap),
+    /* TODO: check socketcall arguments */
+    SCMP_SYS(socketcall),
+#endif
   };
 
   const int denied_any[] = {
