--- dropbear-2025.88/src/mlkem768.c.orig	2025-05-07 10:30:00.000000000 -0400
+++ dropbear-2025.88/src/mlkem768.c	2025-10-29 16:16:56.736435055 -0400
@@ -46,10 +46,7 @@
 #include "dbutil.h"
 #include "compat.h"
 
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wunused-parameter"
 #include "libcrux_mlkem768_sha3.h"
-#pragma GCC diagnostic pop
 
 #include "mlkem768.h"
 #include "dbrandom.h"
--- dropbear-2025.88/libtommath/bn_prime_tab.c.orig	2025-05-07 10:30:00.000000000 -0400
+++ dropbear-2025.88/libtommath/bn_prime_tab.c	2025-10-29 16:16:26.808435162 -0400
@@ -44,7 +44,7 @@
 #endif
 };
 
-#if defined(__GNUC__) && __GNUC__ >= 4
+#if defined(__GNUC__) && __GNUC__ >= 5
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 const mp_digit *s_mp_prime_tab = ltm_prime_tab;
