diff -uNr mini-snmpd-1.6/configure.ac mini-snmpd-1.6.new/configure.ac
--- mini-snmpd-1.6/configure.ac	2021-01-08 03:55:55.428494933 +0000
+++ mini-snmpd-1.6.new/configure.ac	2021-01-08 03:57:31.497576560 +0000
@@ -1,7 +1,7 @@
 AC_INIT([mini-snmpd], [1.6], [https://github.com/troglobit/mini-snmpd/issues],,
 	[https://troglobit.com/projects/mini-snmpd/])
 AC_CONFIG_AUX_DIR(aux)
-AM_INIT_AUTOMAKE([1.11 foreign subdir-objects])
+AM_INIT_AUTOMAKE([1.10 foreign subdir-objects])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
 AC_CONFIG_SRCDIR([mini-snmpd.c])
