--- dropbear-2025.89/src/auth.h
+++ dropbear-2025.89/src/auth.h
@@ -127,6 +127,9 @@ struct AuthState {
 	struct timespec auth_starttime; /* Server only, time of receiving current 
 									SSH_MSG_USERAUTH_REQUEST */
 
+	 /* Count of public keys attempted, limited by MAX_PUBKEY_QUERIES */
+	unsigned int serv_pubkey_query_count;
+
 	/* These are only used for the server */
 	uid_t pw_uid;
 	gid_t pw_gid;
--- a/src/default_options.h
+++ b/src/default_options.h
@@ -303,6 +303,24 @@ group1 in Dropbear server too */
 /* -T server option overrides */
 #define MAX_AUTH_TRIES 10
 
+/* Maximum number of public key queries per session.
+ * Public key queries
+ * aren't a risk for brute forcing authentication, but can be a
+ * user enumeration/privacy concern if an attacker attempts
+ * to iterate known public keys such as those published by github.
+ *
+ * This limit has a trade-off. Having a smaller limit reduces the
+ * number of legitimate public keys that can be presented
+ * by a client/ssh agent.
+ * Assuming a 100ms session establishment time,
+ * MAX_UNAUTH_CLIENTS * MAX_PUBKEY_QUERIES / 0.1 = 750 queries/sec.
+ * That is still a a risk against a single host,
+ * but this limit may deter internet-wide scanning.
+ *
+ * If -T argument or MAX_AUTH_TRIES is larger that will be used instead.
+*/
+#define MAX_PUBKEY_QUERIES 15
+
 /* Change server process to user privileges after authentication. */
 #ifndef DROPBEAR_SVR_DROP_PRIVS
 /* Default is enabled. Should only be disabled if platforms are incompatible */
--- dropbear-2025.89/src/svr-authpubkey.c
+++ dropbear-2025.89/src/svr-authpubkey.c
@@ -165,7 +165,16 @@ void svr_auth_pubkey(int valid_user) {
 	}
 
 	if (auth_failure) {
-		send_msg_userauth_failure(0, 0);
+		/* MAX_PUBKEY_QUERIES allows a greater limit of pubkey queries
+		 * than the standard maxauthtries.
+		 * Start counting failures (incrfail) only when it's reaching
+		 * the limit.
+		 */
+		unsigned int free_query_limit = 0;
+			MAX(0, (int)svr_opts.maxauthtries - MAX_PUBKEY_QUERIES);
+		int incrfail = ses.authstate.serv_pubkey_query_count > free_query_limit;
+		send_msg_userauth_failure(0, incrfail);
+		ses.authstate.serv_pubkey_query_count++;
 		goto out;
 	}
 
